compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -977,7 +977,7 @@ object Denotations {
           true
         case MethodNotAMethodMatch =>
           // Java allows defining both a field and a zero-parameter method with the same name
-          !ctx.erasedTypes && !(symbol.is(JavaDefined) && other.symbol.is(JavaDefined))
+          !ctx.erasedTypes && !other.symbol.is(JavaDefined)
```

Comment:
`a.matches(b)` and `b.matches(a)` should always produce the same result (it should be a symmetrical relation, like equality), so I don't think this is a good way to fix the problem.

Reply:
Resolved. Ready for review. @smarter